test/e2e_node/memory_manager_test.go

```diff
@@ -606,27 +720,13 @@ var _ = SIGDescribe("Memory Manager", "[LinuxOnly]", framework.WithDisruptive(),
 				testPod = e2epod.NewPodClient(f).Create(ctx, testPod)
 
 				ginkgo.By("Checking that pod failed to start because of admission error")
-				gomega.Eventually(ctx, func() bool {
+				gomega.Eventually(ctx, func(g gomega.Gomega) {
 					tmpPod, err := e2epod.NewPodClient(f).Get(ctx, testPod.Name, metav1.GetOptions{})
 					framework.ExpectNoError(err)
 
-					if tmpPod.Status.Phase != v1.PodFailed {
-						return false
-					}
-
-					if tmpPod.Status.Reason != "UnexpectedAdmissionError" {
-						return false
-					}
-
-					if !strings.Contains(tmpPod.Status.Message, "Allocate failed due to [memorymanager]") {
-						return false
-					}
-
-					return true
-				}, time.Minute, 5*time.Second).Should(
-					gomega.BeTrueBecause(
-						"the pod succeeded to start, when it should fail with the admission error",
-					))
+					g.Expect(tmpPod.Status.Phase).To(gomega.Equal(v1.PodFailed))
+					g.Expect(tmpPod.Status.Message).To(gomega.ContainSubstring("sum of exclusive container memory requests equals pod budget"))
+				}, time.Minute, 5*time.Second).Should(gomega.Succeed())
```

Comment:
same comment about using `WithTimeout` and `WithPolling` in new code, please

Reply:
thanks, didn't see yet the update though, please check

Reply:
I was missing this change in non pod-level resource managers related tests, done.